File: KRD.RepoBrowser.Web/Scripts/ViewModels/ChangesetFilter.js
var ChangesetFilter = (function () {
  function ChangesetFilter() {
    var _this = this;
    this.TimestampFrom = ko.observable('').subscribeTo('dateFromTopic');
    this.TimestampTo = ko.observable('').subscribeTo('dateToTopic');
    this.Usernames = ko.observableArray([]).subscribeTo('userNamesTopic');
    this.RepositoryNames = ko.observableArray([]).subscribeTo('repositoryNamesTopic');
    this.BranchNames = ko.observableArray([]).subscribeTo('branchNamesTopic');
  }
  return ChangesetFilter;
})();

File: CommitLab.Data/Query/Filters/ChangesetFilter.cs
using System;
using System.Collections.Generic;

namespace CommitLab.Data.Query.Filters
{
  public class ChangesetFilter
  {
    public ChangesetFilter()
    {
      Usernames = new List<string>();

      RepositoryNames = new List<string>();

      BranchNames = new List<string>();
    }

    public DateTime? TimestampFrom { get; set; }

    public DateTime? TimestampTo { get; set; }

    public IList<string> Usernames { get; set; }

    public IList<string> RepositoryNames { get; set; }

    public IList<string> BranchNames { get; set; }
  }
}
